### PR TITLE
Code is changed to use AstractBotJob save method; Added workaround to ignore limit of saves

### PR DESCRIPTION
**Pull request by alex-korobko**

### CODE REVIEW

File: add_nyt_review_link.py [hunk 1]
```diff
@@ -114,11 +118,10 @@ def __process_found_bestseller_edition(self, bstslr_record_isbn,
                 'for the work {} of the edition {}'
                     .format(bstslr_edition.work.olid,
                             bstslr_record_isbn))
-            self.__add_link(work, link_struct)
-            work.save(comment)
```
Comment (RayBB) [inline add_nyt_review_link.py:118]: If we don't specify the comment here where does it get specified?

File: add_nyt_review_link.py [hunk 2]
```diff
@@ -191,8 +195,15 @@ def run(self) -> None:  # overwrites the AbstractBotJob run method
         comment = 'Add NYT review links'
         with open(self.args.file, 'r') as fin:
             review_record_array = json.load(fin)
-            for review_record in tqdm(review_record_array, unit='reviews'):
```
Comment (RayBB) [inline add_nyt_review_link.py:194]: I think you may have accidentally reverted changes I made here to simplify the usage of tqdm.

File: add_nyt_review_link.py [hunk 3]
```diff
@@ -69,6 +69,10 @@ def __add_link(self, work, link_struct) -> None:
                     .format(work.olid))
         except AttributeError:
             work.links = [link_struct]
+            self.logger.info(
```
Comment (RayBB) [inline add_nyt_review_link.py]: I don't think we need this log as it is logging expected behavior

File: add_nyt_bestseller_tag.py [hunk 4]
```diff
@@ -119,41 +117,39 @@ def __process_bestseller_group_record(self, bestseller_group_record,
...
                 raise
             except:
                 self.logger.exception('Failed to process ISBN {}'
                                       .format(bstslr_record_isbn))
-                job_results['isbns_failed'] = job_results['isbns_failed'] + 1
+                job_results['isbns_failed'] += 1
 
     def run(self) -> None:  # overwrites the AbstractBotJob run method
         self.dry_run = self.args.dry_run
+        self.limit = None
```
Comment (RayBB) [inline add_nyt_bestseller_tag.py:144]: Hm is this what we have to do to get around limit when doing self.save instead of work.save?

**Review by RayBB — CHANGES_REQUESTED** [review]
I'm glad you figured out the right way to save. Just need to clean up a few other things

File: add_nyt_review_link.py (hunk 1 above)
Comment (alex-korobko) [reply add_nyt_review_link.py:118]: I don't see a way to provide a comment if the save function is used. That's why I had to remove the comment altogether.

File: add_nyt_review_link.py (hunk 2 above)
Comment (alex-korobko) [reply add_nyt_review_link.py:194]: yeah, sorry, I did not realize you did the changes after I explained why I use more complex approach with
 manual update https://github.com/RayBB/openlibrary-nyt-bot-temp/pull/6#discussion_r684306216 If you are OK to see the progress bar every other line in logs, I'm OK to put your, simplified version back.

File: add_nyt_review_link.py (hunk 3 above)
Comment (alex-korobko) [reply add_nyt_review_link.py]: Yeah, but as we don't have comments calling the save function now, maybe we could try to compensate via logging in this place? I have no strong opinion about that though, so I would remove that info and probably another one that logs happy path several lines above.

File: add_nyt_bestseller_tag.py (hunk 4 above)
Comment (alex-korobko) [reply add_nyt_bestseller_tag.py:144]: Yes, the self.save function checks limit if it is not a dry_run, so I had to reset that check (alternatively for reviews we could probably set it to the len of the reviews array)

File: add_nyt_review_link.py (hunk 1 above)
Comment (RayBB) [reply add_nyt_review_link.py:118]: Looks like we may need to use a lambda function like they do here. 
What do you think?
https://github.com/internetarchive/openlibrary-bots/blob/8f5506ce2708055ba2c66037e72f3ad51af97981/isbnbot/normalize_isbns.py#L115

File: add_nyt_review_link.py (hunk 2 above)
Comment (RayBB) [reply add_nyt_review_link.py:194]: I think it's fine to see it every time for the sake of simpler code.

File: add_nyt_review_link.py (hunk 3 above)
Comment (RayBB) [reply add_nyt_review_link.py]: I do think it's helpful for us to log which works are done as they're done. Perhaps we can just make the message the save as the above one.

File: add_nyt_review_link.py (hunk 1 above)
Comment (alex-korobko) [reply add_nyt_review_link.py:118]: Good point! I was not aware Python has closures. Modified the code.

**Comment by RayBB** [comment]
@alex-korobko merged!

**Pull request merged**